packages/@aws-cdk-testing/framework-integ/test/aws-rds/test/integ.instance-with-parameter-group.ts

```diff
@@ -0,0 +1,38 @@
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as cdk from 'aws-cdk-lib';
+import * as rds from 'aws-cdk-lib/aws-rds';
+import { IntegTest } from '@aws-cdk/integ-tests-alpha';
+
+const app = new cdk.App();
+const stack = new cdk.Stack(app, 'aws-cdk-rds-instance-with-rds-parameter-group', {
+  terminationProtection: false,
+});
+
+const vpc = new ec2.Vpc(stack, 'VPC', { maxAzs: 2, restrictDefaultSecurityGroup: false });
+
+const parameterGroup = new rds.ParameterGroup(stack, 'ParameterGroup', {
+  engine: rds.DatabaseInstanceEngine.postgres({ version: rds.PostgresEngineVersion.VER_15_2 }),
+  description: 'desc',
+  removalPolicy: cdk.RemovalPolicy.DESTROY,
+});
+
+new rds.DatabaseInstance(stack, 'Instance', {
+  engine: rds.DatabaseInstanceEngine.postgres({ version: rds.PostgresEngineVersion.VER_15_2 }),
+  vpc,
+  multiAz: false,
+  publiclyAccessible: true,
+  iamAuthentication: true,
+  parameterGroup,
+  removalPolicy: cdk.RemovalPolicy.DESTROY,
```

Comment:
The original issue was not working when `removalPolicy: RemovalPolicy.RETAIN` was set. I know for an integ test we can't leave behind the database but I just want to confirm that either:
1.  The issue wasn't specific to the `RETAIN` removal policy but rather any removal policy set for the databaseInstance or
2. If the issue was specific to the `RETAIN` removal policy that you were able to verify locally that the stack was sucessfully destroyed when the removal policy was `RETAIN`.

If the second case is true maybe we could update this integ test to use `RemovalPolicy.RETAIN` and verify that the stack was destroyed before cleaning up the databaseInstance?

Reply:
Yes, this issue is not specific to `RETAIN` rather setting any removal policy. I confirmed <2> by deploying a test stack with setting DBInstance and ParameterGroup to `RETAIN` and deleting the stack. I can confirm that I'm no longer encountering the issue described in the issue #22141.

We can't update the integ test to use `RETAIN` as we won't be able to cleanup the retained resources once the stack is deleted, which will cause leftover resources.